addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/handler/thermostat/NATherm1Handler.java

```diff
@@ -34,79 +33,62 @@
+        super(thing, NATherm1Configuration.class);
+
     }
 
     @Override
-    protected void updateChannels() {
-        try {
-            NAThermStateResponse thermState = bridgeHandler.getThermostatApi().getthermstate(getParentId(), getId());
-            thermStateBody = thermState.getBody();
-
-            NADeviceListBody deviceList = bridgeHandler.getThermostatApi().devicelist(actualApp, getParentId(), false)
-                    .getBody();
-            for (NAModule module : deviceList.getModules()) {
-                if (module.getId().equalsIgnoreCase(getId())) {
-                    this.module = module;
-                    super.updateChannels();
-                }
-            }
-        } catch (Exception e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, e.getMessage());
-        }
-
+    public void updateChannels(NetatmoBridgeHandler bridgeHandler, NAModuleAdapter module) {
+        measures = bridgeHandler.getThermostatApi().getmeasure(configuration.getParentId(), "max",
+                new CSVParams(measuredChannels), module.getId(), null, null, 1, true, true);
+        super.updateChannels(bridgeHandler, module);
     }
 
     @Override
     protected State getNAThingProperty(String chanelId) {
-        switch (chanelId) {
-            case CHANNEL_SETPOINT_MODE:
-                if (thermStateBody.getSetpoint() != null) {
-                    return new StringType(thermStateBody.getSetpoint().getSetpointMode());
-                } else {
-                    return null;
-                }
-            case CHANNEL_SETPOINT_TEMP:
-                return new DecimalType(thermStateBody.getMeasured().getSetpointTemp());
-            case CHANNEL_BOILER_ON:
-                return new DecimalType(dashboard.getBoilerOn());
-            case CHANNEL_BOILER_OFF:
-                return new DecimalType(dashboard.getBoilerOff());
-            default:
-                return super.getNAThingProperty(chanelId);
+        NAThermostat thermostat = module.getThermostat();
+        if (thermostat != null) {
+            switch (chanelId) {
+                case CHANNEL_TEMPERATURE:
+                    return toDecimalType(thermostat.getMeasured().getTemperature());
+                case CHANNEL_SETPOINT_TEMP:
+                    return toDecimalType(thermostat.getMeasured().getSetpointTemp());
+                case CHANNEL_TIMEUTC:
+                    return toDateTimeType(thermostat.getMeasured().getTime());
+                case CHANNEL_SETPOINT_MODE:
+                    if (thermostat.getSetpoint() != null) {
+                        return new StringType(thermostat.getSetpoint().getSetpointMode());
+                    }
+            }
         }
+        return super.getNAThingProperty(chanelId);
     }
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
+        NetatmoBridgeHandler bridgeHandler = (NetatmoBridgeHandler) getBridge().getHandler();
```

Comment:
This is done by default if the switch is not satisfied : updateChannels(configuration.getParentId())